src/actionTypes.js

```diff
@@ -341,35 +341,37 @@ type EventMessageDeleteAction = {|
   type: typeof EVENT_MESSAGE_DELETE,
   messageIds: number[],
 |};
+type TopicLinks = $ReadOnlyArray<{| +text: string, +url: string |}> | $ReadOnlyArray<string>;
+// This is current to feature level 109.
 type EventUpdateMessageAction = {|
   ...ServerEvent,
   type: typeof EVENT_UPDATE_MESSAGE,
-  edit_timestamp: number,
+  user_id: UserId,
   message_id: number,
-  // TODO is it really right that just one of the orig_* is optional?
-  orig_content: string,
-
-  // TODO: The doc for this field isn't yet correct; it turns out that
-  // the question of whether `orig_subject` is present or not is
-  // complicated; see discussion at
-  // https://chat.zulip.org/#narrow/stream/206-zulip-terminal/topic/subject.20always.20present.20in.20event/near/1098954.
-  //
-  // We can be pretty sure of a few things, though:
-  // - it will not be present if the message doesn't have a topic
-  //   (i.e., if it's a private message)
-  // - it's guaranteed to be present if the topic did indeed change
-  // - it will never be an empty string, because the server doesn't
-  //   accept the empty string for a message's topic; it requires
-  //   clients to specify something like `(no topic)` if no topic is
-  //   desired.
+  message_ids: $ReadOnlyArray<number>,
+  edit_timestamp: number,
+  stream_name?: string,
+  stream_id?: number,
+  new_stream_id?: number,
+  propagate_mode: 'change_one' | 'change_later' | 'change_all',
   orig_subject?: string,
+  subject: string,
 
+  // TODO(server-3.0): Added in feat. 46, replacing subject_links
```

Comment:
I think this TODO can be dropped, because there's not actually anything to do at this spot when dropping support for these old servers: the field is optional even in the latest version.